DFT/jesus_dft.py

```diff
@@ -574,3 +575,340 @@ def show_signal_cosines(self):
+                "ƒ = " + v_freq + " Hz",
+                font=REDUCIBLE_FONT,
+                t2f={v_freq: REDUCIBLE_MONO},
+            ).scale(0.8)
+
+            phi_eq = MathTex(r"\phi = ")
+            tex_phase_n = Text(
+                v_phase + "º",
+                font=REDUCIBLE_FONT,
+                t2f={v_phase: REDUCIBLE_MONO},
+            ).scale(0.8)
+            tex_phase = VGroup(phi_eq, tex_phase_n).arrange(RIGHT)
+
+            tex_amplitude = Text(
+                "A = " + v_amplitude,
+                font=REDUCIBLE_FONT,
+                t2f={v_amplitude: REDUCIBLE_MONO},
+            ).scale(0.8)
+
+            tex_b = Text(
+                "b = " + v_b,
+                font=REDUCIBLE_FONT,
+                t2f={v_b: REDUCIBLE_MONO},
+            ).scale(0.8)
+
+            text_group = (
+                VGroup(tex_frequency, tex_phase, tex_amplitude, tex_b)
+                .arrange(DOWN, aligned_edge=LEFT)
+                .scale(0.6)
+                .to_corner(UL)
+            )
+            return text_group
+
+        def change_phase_redraw():
+            phase_ch_cos = get_cosine_func(
+                amplitude=vt_amplitude.get_value(),
+                freq=vt_frequency.get_value(),
+                phase=vt_phase.get_value(),
+                b=vt_b.get_value(),
+            )
+            _, phase_ch_cos_mob = plot_time_domain(phase_ch_cos, t_max=t_max)
+            return phase_ch_cos_mob.scale(0.6).shift(UP)
+
+        af_matrix = get_analysis_frequency_matrix(
+            N=n_samples, sample_rate=sample_frequency, t_max=t_max
+        )
+
+        rect_scale = 0.1
+
+        def updating_transform_redraw():
+            signal_function = get_cosine_func(
+                amplitude=vt_amplitude.get_value(),
+                freq=vt_frequency.get_value(),
+                phase=vt_phase.get_value(),
+                b=vt_b.get_value(),
+            )
+
+            sampled_signal = np.array(
+                [
+                    signal_function(v)
+                    for v in np.linspace(t_min, t_max, num=n_samples, endpoint=False)
+                ]
+            ).reshape(-1, 1)
+
+            # matrix transform
+            mt = apply_matrix_transform(sampled_signal, af_matrix)
+
+            rects = (
+                VGroup(
+                    *[
+                        VGroup(
+                            Rectangle(
+                                color=REDUCIBLE_VIOLET, width=0.3, height=f * rect_scale
+                            ).set_fill(REDUCIBLE_VIOLET, opacity=1),
+                            Text(str(i), font=REDUCIBLE_MONO).scale(0.4),
+                        ).arrange(DOWN)
+                        for i, f in enumerate(mt.flatten()[: mt.shape[0] // 2])
+                    ]
+                )
+                .arrange(RIGHT, aligned_edge=DOWN)
+                .scale(0.6)
+                .move_to(DOWN * 3.4, aligned_edge=DOWN)
+            )
```

Comment:
Highly suggest moving this to a function in dft_utils.py -- it will be useful in a variety of scenes

Reply:
Could be nice to pass in a `sampled_signal`, `af_matrix` with the required rectangle parameters and get something that looks like a bar chart version of it.